### PR TITLE
fix(vrl)!: Make remainder fallible

### LINKED ISSUE #11153: Calculating the remainder with a divisor of 0 in VRL panics
This VRL will cause a panic:

```
3 % 0
```

```
$ 3 % 0
thread 'main' panicked at 'attempt to calculate the remainder with a divisor of zero', lib/vrl/compiler/src/value/arithmetic.rs:121:36
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

This expression should be fallible, and return an error.

### PR DESCRIPTION
**Pull request by StephenWakely**
Fixes #11153 

This makes remainder fallible, so any divide by errors need to be handled.

It should work the same as divide in that dividing by  constant non-zero value will not be fallible.

### CODE REVIEW

**Review by jszwedko — APPROVED** [review]
Good find.

File: lib/vrl/compiler/src/expression/op.rs [hunk 1]
```diff
@@ -191,7 +191,7 @@ impl Expression for Op {
                 let td = TypeDef::float();
 
                 // Division is infallible if the rhs is a literal normal float or integer.
-                match self.rhs.as_value() {
+                match dbg!(self.rhs.as_value()) {
```
Comment (JeanMertz) [inline lib/vrl/compiler/src/expression/op.rs:194]: woops

**Review by JeanMertz — APPROVED** [review]

File: lib/vrl/compiler/src/expression/op.rs (hunk 1 above)
Comment (jszwedko) [reply lib/vrl/compiler/src/expression/op.rs:194]: Hrm, I'm wondering why clippy didn't catch this with `-Dclippy::dbg_macro` in `.cargo/config`.

Comment (jszwedko) [reply lib/vrl/compiler/src/expression/op.rs:194]: Figured out the issue: https://github.com/vectordotdev/vector/pull/11674

**Pull request merged**